pandas/core/dtypes/astype.py

```diff
@@ -145,11 +145,13 @@ def astype_nansafe(
         if is_datetime64_dtype(dtype):
             from pandas import to_datetime
 
```

Comment:
simpler idea.  set `dtype = np.dtype("M8[ns]")` here, since we're going to end up doing that anyway

Reply:
Have set dtype "M8[ns]"